commands/audit/audit_test.go

```diff
@@ -933,3 +933,125 @@ func TestCreateResultsContext(t *testing.T) {
+
+	testCases := []struct {
+		name                  string
+		sourceBranchResults   *results.SecurityCommandResults
+		expectedSbom          results.Sbom
+		expectedApiCallsCount map[string]int
+	}{
+		{
+			name:                "Source branch scan",
+			sourceBranchResults: nil,
+			expectedSbom: results.Sbom{
+				Components: []results.SbomEntry{
+					{
+						Component: "pip",
+						Version:   "25.0.1",
+						Type:      "Python",
+						XrayType:  "pypi",
+						Direct:    true,
+					},
+					{
+						Component: "pyyaml",
+						Version:   "5.2",
+						Type:      "Python",
+						XrayType:  "pypi",
+						Direct:    true,
+					},
+					{
+						Component: "werkzeug",
+						Version:   "1.0.1",
+						Type:      "Python",
+						XrayType:  "pypi",
+						Direct:    true,
+					},
+				},
+			},
+			expectedApiCallsCount: map[string]int{},
+		},
+		{
+			name: "Target branch scan",
+			sourceBranchResults: &results.SecurityCommandResults{
+				Targets: []*results.TargetResults{
+					{
+						ScanTarget: results.ScanTarget{
+							Target:     tempDirPath,
+							Name:       "",
+							Technology: "pip",
+						},
+						Sbom: results.Sbom{
+							Components: []results.SbomEntry{
+								{
+									Component: "pip",
+									Version:   "26.0",
+									Type:      "Python",
+									XrayType:  "pypi",
+								},
+								{
+									Component: "pyyaml",
+									Version:   "5.2",
+									Type:      "Python",
+									XrayType:  "pypi",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedSbom: results.Sbom{
+				Components: []results.SbomEntry{
+					{
+						Component: "pip",
+						Version:   "26.0",
+						Type:      "Python",
+						XrayType:  "pypi",
+						Direct:    true,
+					},
+				},
+			},
+			expectedApiCallsCount: map[string]int{
+				validations.GraphScanPostAPI: 1,
+				validations.GraphScanGetAPI:  1,
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			testParams := validations.MockServerParams{
+				XrayVersion:   utils.EntitlementsMinVersion,
+				XscVersion:    services.ConfigProfileMinXscVersion,
+				ApiCallCounts: map[string]int{},
+			}
+			mockServer, serverDetails := validations.XrayServer(t, &testParams)
+			defer mockServer.Close()
+
+			auditBasicParams := (&utils.AuditBasicParams{}).
+				SetServerDetails(serverDetails).
+				SetXrayVersion(utils.EntitlementsMinVersion).
+				SetXscVersion(services.ConfigProfileMinXscVersion).
+				SetOutputFormat(format.SimpleJson).
+				SetUseJas(false)
+
+			auditParams := NewAuditParams().
+				SetWorkingDirs([]string{tempDirPath}).
+				SetMultiScanId(validations.TestMsi).
+				SetGraphBasicParams(auditBasicParams).
+				SetResultsContext(results.ResultContext{IncludeVulnerabilities: true}).
+				SetDiffMode(true).
+				SetResultsToCompare(tc.sourceBranchResults)
+
+			auditParams.SetWorkingDirs([]string{tempDirPath}).SetIsRecursiveScan(true)
```

Comment:
```suggestion
			auditParams.SetIsRecursiveScan(true)
```
already set `SetWorkingDirs` above